### PR TITLE
Implement external keystore

### PR DESCRIPTION
**Pull request by majecty**
I run integration tests with injected external keystore.
It passes

### CODE REVIEW

File: src/key/ExternalKeyStore.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        return response.result;
+    }
+
+    async getPK(params: { hash: string; }): Promise<string> {
+        const response = await rp.get(`${this.keystoreURL}/api/pkhs/${params.hash}`, {
+            json: true
+        });
+
+        if (!response.success) {
+            console.dir(response);
```
Comment (joojis) [inline src/key/ExternalKeyStore.ts]: I guess you forgot to remove this line.

File: src/key/ExternalKeyStore.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import * as _ from "lodash";
+
+import { KeyStore } from "./KeyStore";
+import * as rp from "request-promise";
```
Comment (joojis) [inline src/key/ExternalKeyStore.ts]: Order of imports

File: src/key/ExternalKeyStore.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+export class ExternalKeyStore implements KeyStore {
+    keystoreURL: string;
+
+    constructor(keystoreURL: string) {
+        this.keystoreURL = keystoreURL;
+    }
```
Comment (joojis) [inline src/key/ExternalKeyStore.ts]: I hope for there is a way to check whether the server is available.

**Comment by majecty** [comment]
@joojis I fixed

**Pull request merged**